### PR TITLE
LibSoftGPU: east const and const-correctness in Clipper

### PR DESCRIPTION
**Pull request by ldm5180**

### CODE REVIEW

**Comment by Hendiadyoin1** [comment]
Is ClipPlane trivially copyable?
If no could you mark it as const& and not just const?
Otherwise is it not really necessary to be const-correct to this degree, but it's just my opinion.

**Comment by ldm5180** [comment]
> Is ClipPlane trivially copyable? If no could you mark it as const& and not just const? Otherwise is it not really necessary to be const-correct to this degree, but that's just my opinion.

Yes. It is an enum which has an underlying type of u8. While not really being necessary, it might as well be made correct while cleaning this up, right?

File: Userland/Libraries/LibSoftGPU/Clipper.h [hunk 1]
```diff
@@ -49,8 +49,8 @@ class Clipper final {
     void clip_triangle_against_frustum(Vector<Vertex>& input_vecs);
 
 private:
-    bool point_within_clip_plane(FloatVector4 const& vertex, ClipPlane plane);
-    Vertex clip_intersection_point(Vertex const& vec, Vertex const& prev_vec, ClipPlane plane_index);
+    bool point_within_clip_plane(FloatVector4 const& vertex, ClipPlane const plane) const;
+    Vertex clip_intersection_point(Vertex const& vec, Vertex const& prev_vec, ClipPlane const plane_index) const;
```
Comment (alimpfard) [inline Userland/Libraries/LibSoftGPU/Clipper.h:53]: This seems pointless, does this do something?
It seems to be a net negative from what I can tell:
- leaking aspects of the implementation of these functions
- making it annoying for the hypothetical future person that ends up wanting to modify their copy of this ClipPlane
- ...is a change...that does nothing...?

(making the member function const is 👍 tho)

Comment (ldm5180) [reply Userland/Libraries/LibSoftGPU/Clipper.h:53]: This is going to be moved to be a template argument in a future commit (see PR #11682 for context). I believe making this `const` demonstrates a little bit of the ability to move it to be a static compile-time constant when the time comes. It eliminates a potential concern in the next PR about the function ever modifying it.

Comment (alimpfard) [reply Userland/Libraries/LibSoftGPU/Clipper.h:53]: const enum parameter still feels weird, but 🤷

**Review by linusg — COMMENTED** [review]
`clipper.[h,cpp] ` should have an uppercase `C`, code LGTM.

**Pull request merged**